### ISSUE TITLE
Streaming RPC server is not aware of client disconnect on Android

### ISSUE THREAD

**Issue by AsturaPhoenix** [vanadium/issues#1361]
I have a bidirectional streaming RPC between two Android devices. When the Vanadium client device goes into airplane mode, it properly detects that the connection is broken. However, the server device continues to be able to send on its side of the stream, completing successfully and not erroring, for many minutes until the channelTimeout expires (~15 mins).

If the server device goes into airplane mode, both devices detect that the connection is broken.

**Comment by ghost** [comment]
Yeah, other than having a stricter connection timeout I'm not sure what we can do about this.  The kernel isn't telling us that the connection is closed, and write syscalls are succeeding.

I'll think about it, but I wouldn't expect any quick action.

**Comment by AsturaPhoenix** [comment]
I think the stricter connection timeout is the preferred solution for us anyway short of instantly magically knowing that a connection has been broken, since as you told me the connection timeout is actually implemented as a pair of heartbeats anyway. I'm looking into what it will take to do this from the Java side now.

Is there somewhere we should document this caveat? I'd be happy to add it if there's a home for it, or start a doc if not.
